edx_proctoring/views.py

```diff
@@ -3,23 +3,176 @@
+
+    HTTP PUT
+    Updates an existing Exam.
+    PUT data : {
+        "exam_id": 533,
+        "exam_name": "Final",
+        "time_limit_mins": 120,
+        "is_proctored": true,
+        "external_id": 235
+        "is_active": true
+    }
+
+    **PUT data Parameters**
+        see the POST data parameters
+
+    **Response Values**
+        * {'exam_id': ##}, The exam_id of the created Proctored Exam.
+
+
+    HTTP GET
+        ** Scenarios **
+        ?exam_id=533
+        returns an existing exam  object matching the exam_id
+
+        ?course_id=edX/DemoX/Demo_Course&content_id=123
+        returns an existing exam object matching the course_id and the content_id
+    """
+    def post(self, request):
+        """
+        Http POST handler. Creates an exam.
+        """
+        try:
+            exam_id = create_exam(
+                course_id=request.DATA.get('course_id', ""),
+                content_id=request.DATA.get('content_id', ""),
+                exam_name=request.DATA.get('exam_name', ""),
+                time_limit_mins=request.DATA.get('time_limit_mins', ""),
+                is_proctored=True if request.DATA.get('is_proctored', "False").lower() == 'true' else False,
+                external_id=request.DATA.get('external_id', ""),
+                is_active=True if request.DATA.get('is_active', "").lower() == 'true' else False,
+            )
+            return Response({'exam_id': exam_id})
+        except IntegrityError:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={"detail": "Trying to create a duplicate exam."}
+            )
+
+    def put(self, request):
+        """
+        HTTP PUT handler. To update an exam.
+        calls the update_exam
+        """
+        try:
+            exam_id = update_exam(
+                exam_id=request.DATA.get('exam_id', ""),
+                exam_name=request.DATA.get('exam_name', ""),
+                time_limit_mins=request.DATA.get('time_limit_mins', ""),
+                is_proctored=True if request.DATA.get('is_proctored', "False").lower() == 'true' else False,
+                external_id=request.DATA.get('external_id', ""),
+                is_active=True if request.DATA.get('is_active', "").lower() == 'true' else False,
+            )
+            return Response({'exam_id': exam_id})
+        except:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={"detail": "The exam_id does not exist."}
+            )
+
+    def get(self, request):
+        """
+        HTTP GET handler.
+            Scenarios:
+                by exam_id: calls get_exam_by_id()
+                by course_id, content_id: get_exam_by_content_id()
+
+        """
+
+        course_id = request.QUERY_PARAMS.get('course_id', None)
+        content_id = request.QUERY_PARAMS.get('content_id', None)
+        exam_id = request.QUERY_PARAMS.get('exam_id', None)
```

Comment:
In general, we never use query string parameters, but rather put them in the URL paths so that they are passed into the handler.  I'll see if I can find some examples for you.